espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/ActivityOptionsHelpers.kt

```diff
@@ -28,15 +29,18 @@ import android.app.ActivityOptions
  * there is an issue with mapping value format
  */
 
-fun makeActivityOptions(activityOptions: Map<String, String>?): ActivityOptions {
+fun makeActivityOptions(activityOptions: Map<String, Any?>?): ActivityOptions {
     val options = ActivityOptions.makeBasic()
-    val handlersMapping = mapOf<String, (key: String, value: String) -> Unit>(
+    val handlersMapping = mapOf<String, (key: String, value: Any?) -> Unit>(
             "launchDisplayId" to fun(_, value) {
-                options.launchDisplayId = value.toInt()
+                require(String::class.isInstance(value)) {
```

Comment:
we should be able to accept both string and int:

```
try {
  if (String::class.isInstance(value)) {
      options.launchDisplayId = String::class.cast(value).toInt()
      return
  } else if (Number::class.isInstance(value)) {
      options.launchDisplayId = Number::class.cast(value).toInt()
      return
  }
} catch (NumberFormatException: e) {
   // cannot convert to int
   throw IllegalArgumentException...
}
// not string or number passed
throw IllegalArgumentException...
```

Reply:
Thanks a lot, I will add and test it.